frontend/public/components/storage/create-pvc.tsx

```diff
@@ -0,0 +1,368 @@
+  return (
+    <ListDropdown
+      {...props}
+      desc="Storage Classes"
+      resources={resources}
+      selectedKeyKind={kind}
+      placeholder="Select storage class"
+      selectedKey={selectedKey}
+      required={required}
+      namespace={namespace}
+      name={name}
+    />
+  );
+};
+// This form is done a little odd since it is used in both its own page and as
+// a sub form inside the attach storage page.
+export class CreatePvcForm extends React.Component<CreatePvcFormProps, CreatePvcFormState> {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      storageClass: '',
+      pvcName: '',
+      accessMode: 'ReadWriteOnce',
+      requestSizeValue: '1',
+      requestSizeUnit: 'Gi',
+      disableForm: false,
+      showLabelSelectors: false,
+      nameValuePairs: [['', '']],
+      radios: [
+        {
+          value: 'ReadWriteOnce',
+          title: 'Single User (RWO)',
+        },
+        {
+          value: 'ReadWriteMany',
+          title: 'Shared Access (RWX)',
+        },
+        {
+          value: 'ReadOnlyMany',
+          title: 'Read Only (ROX)',
+        },
+      ],
+      dropdownUnits: {
+        Mi: 'MiB',
+        Gi: 'GiB',
+        Ti: 'TiB',
+      },
+    };
+  }
+
+  handleChange: React.ReactEventHandler<HTMLInputElement> = event => {
+    // this handles pvcName, accessMode, size
+    const { name, value } = event.currentTarget;
+    this.setState({ [name]: value } as any, () => {
+      this.onChange();
+    });
+  };
+
+  handleNameValuePairs = ({ nameValuePairs }) => {
+    this.setState({ nameValuePairs }, () => {
+      this.onChange();
+    });
+  };
+
+  handleStorageClass = storageClass => {
+    this.setState({ storageClass }, () => {
+      this.onChange();
+    });
+  };
+
+  handleClear = () => {
+    this.setState({ storageClass: '' }, () => {
+      this.onChange();
+    });
+  };
+
+  toggleShowLabelSelectors = () => {
+    this.setState({ showLabelSelectors: !this.state.showLabelSelectors });
+  };
+
+  onFormDisable = disableForm => {
+    this.setState({ disableForm });
+  };
+
+  nameValueArrayToObject = nvpArray => {
+    return _.reduce(nvpArray, (acc, [key, value]) => {
+      return key && value ? { ...acc, [key]: value } : acc;
+    }, {});
+  };
+
+  onChange = () => {
+    return this.props.onChange(this.updatePvc());
+  }
+
+  updatePvc = () => {
+    const { namespace } = this.props;
+    const { pvcName, accessMode, requestSizeValue, requestSizeUnit, storageClass, nameValuePairs } = this.state;
+    const valuePairs = this.nameValueArrayToObject(nameValuePairs);
+
+    const obj = {
+      apiVersion: 'v1',
+      kind: 'PersistentVolumeClaim',
+      metadata: {
+        name: pvcName,
+        namespace: namespace,
+        annotations: {},
+      },
+      spec: {
+        accessModes: [accessMode],
+        storageClassName: (storageClass && storageClass !== 'No Storage Class') ? storageClass : '',
```

Comment:
We shouldn't be checking for "No Storage Class" here... We should fix it so the dropdown returns an empty value if nothing is selected, however we need to do that.